planner/core/preprocess.go

```diff
@@ -831,9 +831,11 @@ func checkColumn(colDef *ast.ColumnDef) error {
 		// https://dev.mysql.com/doc/refman/8.0/en/floating-point-types.html
 		if tp.Decimal == -1 {
 			if tp.Tp == mysql.TypeDouble {
-				if tp.Flen != -1 {
-					return types.ErrSyntax.GenWithStackByArgs()
-				}
+				/*
```

Comment:
@AilinKid shall we remove this check and leave it to parser component to check double syntax?

Reply:
Yes I think so, you can remove it and leave a comment here.

Reply:
yes, letting parser do this may be more elegant.